src/Apps/Order/Routes/Shipping/FulfillmentDetailsForm.tsx

```diff
@@ -0,0 +1,830 @@
+              ? "saved_addresses"
+              : "new_address"
+            : "pickup"
+          : null
+        active && values.fulfillmentType === FulfillmentType.SHIP
+        // When not showing the form/creating a new address,
+        // inputs should not be tabbable
+        const tabbableFormValue = (
+          activeForm: typeof showAddressForm
+        ): 0 | -1 => (showAddressForm === activeForm ? 0 : -1)
+
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        const handleSelectSavedAddress = useCallback(
+          (address: ShippingAddressFormValues) => {
+            formikProps.setValues({
+              fulfillmentType: FulfillmentType.SHIP,
+
+              attributes: {
+                ...address,
+                saveAddress: false,
+                addressVerifiedBy: null,
+              },
+            })
+          },
+          [formikProps]
+        )
+
+        return (
+          <Form>
+            {availableFulfillmentTypes.length > 1 && (
+              <>
+                <RadioGroup
+                  data-testid="shipping-options"
+                  onSelect={value => {
+                    formikProps.setFieldValue("fulfillmentType", value)
+                  }}
+                  defaultValue={values.fulfillmentType}
+                >
+                  <Text variant="lg-display" mb="1">
+                    Delivery method
+                  </Text>
+                  <BorderedRadio value="SHIP" label="Shipping" />
+                  <BorderedRadio
+                    value="PICKUP"
+                    label="Arrange for pickup (free)"
+                    data-testid="pickupOption"
+                  >
+                    <Collapse
+                      open={values.fulfillmentType === FulfillmentType.PICKUP}
+                    >
+                      <Text variant="xs" color="black60">
+                        After your order is confirmed, a specialist will contact
+                        you to coordinate pickup.
+                      </Text>
+                    </Collapse>
+                  </BorderedRadio>
+                </RadioGroup>
+                <Spacer y={4} />
+              </>
+            )}
+            {values.fulfillmentType === FulfillmentType.SHIP && (
+              <>
+                <Text variant="lg-display" mb="1">
+                  Delivery address
+                </Text>
+                {/* SAVED ADDRESSES */}
+                <Collapse
+                  data-testid="savedAddressesCollapse"
+                  open={showAddressForm === "saved_addresses"}
+                >
+                  {
+                    // TODO: Maybe this stays in index? We render an error message
+                    // here if shipping quotes are present but empty.
+                    // isArtsyShipping &&
+                    //   shippingQuotes &&
+                    //   shippingQuotes.length === 0 &&
+                    //   renderArtaErrorMessage()
```

Comment:
May need to pass this error message in from the parent Shipping Route (unless this form is to know about shipping quotes) - if so it can probably go below the Text element directly above.